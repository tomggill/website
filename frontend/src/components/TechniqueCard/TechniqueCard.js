import Card from 'react-bootstrap/Card'
import "../../styles/styles.css";

const TechniqueCard = (children) => {
  const {cardTitle, cardText} = children;
  return (
    <Card className='customCard mb-3'>
      <Card.Body>
        <Card.Title className='customCardTitle'>{cardTitle}</Card.Title>
        <Card.Text>
          {cardText}
        </Card.Text>
      </Card.Body>
    </Card>
  );
}
export default TechniqueCard;