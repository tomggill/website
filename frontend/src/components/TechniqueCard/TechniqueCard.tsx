import React from 'react';
import Card from 'react-bootstrap/Card';
import '../../styles/styles.css';

interface TechniqueCardProps {
  cardTitle: string;
  cardText: string;
}
function TechniqueCard(props: TechniqueCardProps) {
  return (
    <Card className="customCard mb-3">
      <Card.Body>
        <Card.Title className="customCardTitle">{props.cardTitle}</Card.Title>
        <Card.Text>
          {props.cardText}
        </Card.Text>
      </Card.Body>
    </Card>
  );
}
export default TechniqueCard;
