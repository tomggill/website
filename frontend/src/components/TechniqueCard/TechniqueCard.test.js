import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import TechniqueCard from './TechniqueCard';

test('Renders TechniqueCard component', () => {
  const cardTitle = 'cardTitleTest';
  render(<TechniqueCard cardTitle={cardTitle} />);
  const linkElement = screen.getByText(cardTitle);
  expect(linkElement).toBeInTheDocument();
});

test('TechniqueCard snapshot test', () => {
  const component = renderer.create(<TechniqueCard cardTitle="cardTitleTest" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
