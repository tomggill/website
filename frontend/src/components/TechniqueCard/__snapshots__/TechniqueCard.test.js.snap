// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechniqueCard snapshot test 1`] = `
<div
  className="customCard mb-3 card"
>
  <div
    className="card-body"
  >
    <div
      className="customCardTitle card-title h5"
    >
      cardTitleTest
    </div>
    <p
      className="card-text"
    />
  </div>
</div>
`;
